packages/strapi-connector-mongoose/lib/mount-models.js

```diff
@@ -291,7 +290,17 @@ module.exports = ({ models, target }, ctx) => {
           }
         }
       });
-    });
+    };
+
+    // Only sync indexes in development env while it's not possible to create complex indexes directly from models
+    // In other environments it will simply create missing indexes (those defined in the models but not present in db)
+    if (strapi.app.env === 'development') {
+      // Ensure indexes are synced with the model, prevent duplicate index errors
+      // Side-effect: Delete all the indexes not present in the model.json
+      Model.syncIndexes(null, handleIndexesErrors);
+    } else {
+      handleIndexesErrors();
+    }
```

Comment:
`handleIndexesErrors` is used in both `if` and `else`. Can it be written once outside? Since `syncIndexes` is sync, it is not needed to use the callback argument.

Reply:
`syncIndexes` is asynchronous (sync is for synchronization here, not synchronous), thus the callback is needed and it can't be written outside

Reply:
syncIndexes is asynchronous (sync is for synchronization here, not synchronous), thus the callback is needed and it can't be written outside.

ref: https://mongoosejs.com/docs/api.html#model_Model.syncIndexes

Reply:
Yeah you're right, I got tricked by the name :p.
Is there a reason to run `handleIndexesErrors` after `syncIndexes` and not before?

Reply:
@Convly ?